containers/withAdminContainer.ts

```diff
@@ -0,0 +1,34 @@
+import { Context } from '../@types/helpers'
+import { isAdmin, isAdminOrThrow } from '../helpers/isAdmin'
+import { withUserContainer } from './withUserContainer'
+import _ from 'lodash'
+
+//use when only checking if user is admin
+export const withAdminContainer =
+  <Type, ArgsType>(
+    resolver: (_parent: void, args: ArgsType, ctx: Context) => Type,
+    errorMessage?: string
+  ) =>
+  async (_parent: void, args: ArgsType, ctx: Context) => {
+    const { req } = ctx
+
+    //check if is admin
+    if (errorMessage) {
+      if (!isAdmin(req)) throw new Error(errorMessage)
+    }
+
+    isAdminOrThrow(req)
+
+    return resolver(_parent, args, ctx)
+  }
+
+export const withAdminUserContainer =
+  <Type, ArgsType>(
+    resolver: (_parent: void, args: ArgsType, ctx: Context) => Type,
+    errorMessage?: string
+  ) =>
+  async (_parent: void, args: ArgsType, ctx: Context) => {
```

Comment:
Can you try

```
export const withAdminUserContainer =
  <Type, ArgsType>(
    resolver: (_parent: void, args: ArgsType, ctx: Context) => Type,
    errorMessage?: string
  ) => withUserContainer(withAdminContainer(resolver, errorMessage))
```

Reply:
@anthonykhoa this didn't work as we need to pass in the` (_parent, args, ctx)` when calling the withUserContainer function. Since `withUserContainer` returns another function.
Like this: `withUserContainer(withAdminContainer(resolver, errorMessage))(_parent, args, ctx)`

Reply:
Just to verify, you tried copying/pasting my example code and it didn't work?

Reply:
So the tests work:

but I keep getting this type error:
![image](https://user-images.githubusercontent.com/77421872/187003046-e509a36a-a23a-4f4c-b931-f697d919aec2.png)

then I tried: 
![image](https://user-images.githubusercontent.com/77421872/187003095-abd7a37b-e140-4ecb-87cc-400fbb60a177.png)

getting the same error.

also tried:
![image](https://user-images.githubusercontent.com/77421872/187007303-c95f5d3c-aab0-42a0-bc58-946ff937ca7a.png)

Reply:
try passing just resolver and errorMessage then

Reply:
doing `...arguments as const` should work too

Reply:
passing in resolver and errorMessage worked, but `...arguments as const` gave the same error